fix(agw): c++ migration of nas/ies files

pruthvihebbani opened the pull request:
fix(agw): c++ migration of nas/ies files

## Summary

Migrated all the files under nas/ies to cpp/hpp and fixed all the compilation errors and warnings

## Test Plan

Executed s1ap integ_tests, bazel build and unit test

pruthvihebbani committed:
Renamed files to cpp/hpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Renamed files to cpp/hpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
resolved compilation errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed compilation errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed compilation errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed compilation errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Resolved undefined reference errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Resolved undefined references

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Merge remote-tracking branch 'upstream/master' into cpp_migration_nas_emm_part2

pruthvihebbani committed:
Converted nas/util files to cpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Addressed self review comments

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed clang errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Adding emm_headers.hpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Migrated files in ies to cpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Migrated files under ies to cpp

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed formatting errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed bazel build errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp:
```diff
@@ -15,18 +15,23 @@
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.hpp"
 #include <stdint.h>
```
Leave the space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeCommand.hpp:
```diff
@@ -18,13 +18,13 @@
 #ifndef FILE_NAS_SECURITY_MODE_COMMAND_SEEN
 #define FILE_NAS_SECURITY_MODE_COMMAND_SEEN
```
Replace with #pragma once

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/NASSecurityModeComplete.hpp:
```diff
@@ -18,8 +18,8 @@
 #ifndef FILE_SECURITY_MODE_COMPLETE_SEEN
 #define FILE_SECURITY_MODE_COMPLETE_SEEN
```
Replace with pragma once

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/SecurityModeReject.hpp:
```diff
@@ -19,9 +19,9 @@
 #define FILE_SECURITY_MODE_REJECT_SEEN
```
Replace with #pragma once

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/ServiceReject.hpp:
```diff
@@ -20,9 +20,9 @@
 
```
Replace with #pragma once for all headers in this PR

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateComplete.hpp:
```diff
@@ -19,8 +19,8 @@
 #define FILE_TRACKING_AREA_UPDATE_COMPLETE_SEEN
 #include <stdint.h>
```
ditto

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateReject.hpp:
```diff
@@ -19,9 +19,9 @@
 #define FILE_TRACKING_AREA_UPDATE_REJECT_SEEN
```
ditto

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/TrackingAreaUpdateRequest.cpp:
```diff
@@ -29,7 +29,7 @@ extern "C" {
 }
 #endif
```
Leave space after

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/emm/msg/UplinkNasTransport.hpp:
```diff
@@ -19,9 +19,9 @@
 #define FILE_UPLINK_NAS_TRANSPORT_SEEN
```
Replace with #pragma once

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/esm/EpsBearerContextDeactivation.cpp:
```diff
@@ -425,7 +425,7 @@ status_code_e eps_bearer_deactivate_t3495_handler(zloop_t* loop, int timer_id,
 
   ue_mm_context_t* ue_mm_context = mme_app_get_ue_context_for_timer(
-      ue_id, "EPS BEARER DEACTIVATE T3495 Timer");
+      ue_id, (char*)"EPS BEARER DEACTIVATE T3495 Timer");
```
Fix clang errors

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/esm/msg/ActivateDefaultEpsBearerContextAccept.hpp:
```diff
@@ -23,7 +23,7 @@
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
Place it outside the extern C

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/esm/msg/EsmInformationResponse.hpp:
```diff
@@ -23,7 +23,7 @@
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
Place it outside extern C

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/esm/msg/ModifyEpsBearerContextAccept.hpp:
```diff
@@ -22,7 +22,7 @@
 extern "C" {
 #endif
-#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.h"
+#include "lte/gateway/c/core/oai/tasks/nas/ies/MessageType.hpp"
```
ditto

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateResult.cpp:
```diff
@@ -15,11 +15,11 @@
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateResult.hpp"
 #include <stdint.h>
```
Leave space after

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.cpp:
```diff
@@ -15,18 +15,23 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/AdditionalUpdateType.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/CsfbResponse.cpp:
```diff
@@ -15,11 +15,11 @@
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/CsfbResponse.hpp"
 #include <stdint.h>
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/DaylightSavingTime.cpp:
```diff
@@ -15,13 +15,13 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/DaylightSavingTime.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/EmmCause.cpp:
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/EmmCause.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/EpsAttachResult.cpp:
```diff
@@ -15,11 +15,11 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/EpsAttachResult.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp:
```diff
@@ -20,23 +20,29 @@
 typedef bstring NasMessageContainer;
 
-#ifdef __cplusplus
+/*#ifdef __cplusplus
```
Remove the commented lines

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/NasMessageContainer.hpp:
```diff
@@ -20,23 +20,29 @@
 int decode_nas_message_container(NasMessageContainer* nasmessagecontainer,
                                  uint8_t iei, uint8_t* buffer, uint32_t len);
-#ifdef __cplusplus
+/*#ifdef __cplusplus
```
Remove the commented lines

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/ProcedureTransactionIdentity.hpp:
```diff
@@ -19,6 +19,8 @@
 #define PROCEDURE_TRANSACTION_IDENTITY_H_
```
Replace with pragma once

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/ProtocolDiscriminator.cpp:
```diff
@@ -15,14 +15,14 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/ProtocolDiscriminator.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentity.cpp:
```diff
@@ -20,7 +20,13 @@
 
 #include "lte/gateway/c/core/oai/common/TLVEncoder.h"
 #include "lte/gateway/c/core/oai/common/TLVDecoder.h"
```
Space

rsarwad commented on lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentityList.cpp:
```diff
@@ -15,13 +15,13 @@
  *      contact@openairinterface.org
  */
 
+#include "lte/gateway/c/core/oai/tasks/nas/ies/TrackingAreaIdentityList.hpp"
```
Space

rsarwad commented on lte/gateway/c/core/oai/test/lib/test_3gpp.cpp:
```diff
@@ -329,4 +330,4 @@ int main(int argc, char** argv) {
   return RUN_ALL_TESTS();
-}
\ No newline at end of file
+}
```
Check whether new line required or not

pruthvihebbani replied:
This is a formatting change only. Somehow as part of clang formatting this got removed and added

rsarwad reviewed (commented):
Left some minor comments

pruthvihebbani committed:
Merge remote-tracking branch 'upstream/master' into cpp_migration_nas_ies

pruthvihebbani committed:
Addressed review comments

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Merge remote-tracking branch 'upstream/master' into cpp_migration_nas_ies

sebathomas committed:
chore(agw): Remove unused code from pydep (#14003)

Remove usages of the tarfile module from pydep. This code is not
used any more.

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

rsarwad committed:
Update gtpv2c_ie_formatter.c

mpfirrmann committed:
chore(agw): s1ap_utils.py is cleaned up (#13854)

* chore(agw): functions are reordered for better readability
* chore(agw): s1ap_utils is cleaned up for readability

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

mpfirrmann committed:
feat(agw): test_sctp_shutdown_while_mme_is_stopped.py is adapted to containerized AGW (#14009)

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

nstng committed:
fix: missing enodebd folders are added to magma debian build (#14007)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

vktng committed:
chore: integ test magma restart waits for pipelined to be initialized (#13980)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>
Co-authored-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

sebathomas committed:
fix(agw): Allow containerized magmad to restart services (#13993)

Add the docker application and the docker socket to the magmad
container setup.

Document that init_system needs to be set in the config.

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

ajay-kashyap committed:
fix(mme): Handling  qosflowFailedForAddorModify in session modification response (#13655)

* fix(mme): Handling  qosflowFailedForAddorModify in session modification response.

Signed-off-by: Ajay-Kashyap <ajay.kashyap@wavelabs.ai>

* Session modification corner case handling

Signed-off-by: Ajay-Kashyap <ajay.kashyap@wavelabs.ai>

Signed-off-by: Ajay-Kashyap <ajay.kashyap@wavelabs.ai>

mpfirrmann committed:
fix(agw): make redis save state after cleaning the db (#13994)

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>
Co-authored-by: Christian Krämer <christian.kraemer@tngtech.com>

dependabot[bot] committed:
chore(deps): bump yajl-ruby in /orc8r/cloud/docker/fluentd_daemon (#12870)

Bumps [yajl-ruby](https://github.com/brianmario/yajl-ruby) from 1.4.2 to 1.4.3.
- [Release notes](https://github.com/brianmario/yajl-ruby/releases)
- [Changelog](https://github.com/brianmario/yajl-ruby/blob/master/CHANGELOG.md)
- [Commits](https://github.com/brianmario/yajl-ruby/commits)

---
updated-dependencies:
- dependency-name: yajl-ruby
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(agw): sctpd needs explicit restart for docker as it is restarted … (#14015)

* fix(agw): sctpd needs explicit restart for docker as it is restarted by systemd autmatically

Signed-off-by: Christian Krämer <christian.kraemer@tngtech.com>

vktng committed:
chore: Publish test results for the Bazel and Debian integration and sudo tests (#13984)

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

Signed-off-by: Krisztián Varga <krisztian.varga@tngtech.com>

nstng committed:
chore: an integ test checks if all magma services are running (#14013)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

nstng committed:
chore: the init system check is less chatty (#14004)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

mpfirrmann committed:
chore(agw): restart of sctpd is adapted (#14000)

* feat(agw): sctpd is stopped before a start via magmad service

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* feat(agw): sctpd is restarted only via restart_services()

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

wolfseb committed:
fix(agw): fix incorrect asserts in agw integ tests (#14012)

* fix(agw): fix incorrect usage of self.assertTrue

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

* chore(agw): change to new pytest assert syntax

This commit changes
* self.assertEqual and self.assertListEqual to assert ... == ...
* self.assertTrue to assert
* self.assertGreater to assert ... > ...
* self.assertGreaterEqual to assert ... >= ...
* self.assertLessEqual to assert ... <= ...
* self.assertIn to assert ... in ...
* self.assertIsNotNone to assert ... is not None

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

* chore(agw): cleanup line lengths from previous commits

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

wolfseb committed:
fix(agw): fix certificate issue for containerized AGW deployment (#13862)

* fix(agw): fix certificate issue for containerized AGW deployment

fixes the mount path for the rootCA.pem certificate, so a containerized deployment
outside the dev VM can still access it

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

* chore: add option to update rootCA.pem in deployment docs

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

nstng committed:
fix: boolean arguments in fabric calls are interpreted correctly (#14030)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

MoritzThomasHuebner committed:
refactor(orc8r): Remove proto.Clone calls (#14029)

* refactor(orc8r): Minor improvements to sql.go

- Remove proto.Clone calls from sql.go
- Added some missing commas to comments
- Changed empty slice creation using literals to nil slice declaration

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Remove proto.Clone calls from sql_test.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Remove proto.Clone calls from client_api.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Remove proto.Clone calls from sql_entity_write_helpers.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Remove proto.Clone calls from storage.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Remove proto.Clone calls from sql_integ_test.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Revert changes in sql_integ_test.go

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* refactor(orc8r): Revert changes to import order

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

Commit:
chore(ci): extracxt magma package version and trigger workflow (#13983)

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

voisey committed:
chore(nms): Improve generic commands functionality (#14014)

* chore(magmad): Improve error handling of generic commands

Work in progress, orc8r (and possibly NMS) needs to be adapted.

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

* Propagate Not Found error to orc8r

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

* Add information about configuration of generic commands

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

* Change default value of parameters field to use shell_params

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

* Handle further error cases

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

* Allow generic command to run with parameters `{"shell_params": []}`

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>
Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>
Co-authored-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

mpfirrmann committed:
fix(agw): all `ROOTCA_PATH` variables are deleted (#14037)

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

Commit:
fix(ci): use correct variable offered by dispatch action (#14038)

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

ganeshg87 committed:
fix(amf): VoNR configuration support (#14039)

Testing:
   verified Registration with NetworkFeature IE support by configuring
   enable/disable

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

Signed-off-by: ganeshg87 <ganesh.gedela@wavelabs.ai>

MoritzThomasHuebner committed:
feat(ci): Move and edit release scripts/workflows (#14011)

* feat(ci): Add docker promote script and workflow

Taken from @ShubhamTatvamasi's repo (https://github.com/ShubhamTatvamasi/magma-artifactory-docker-promote/blob/master/.github/workflows/docker-promote.yml).

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Add helm promote workflow

Taken from @nstng's repo (https://github.com/nstng/magma-artifactory-helm-promote/blob/master/.github/workflows/helm-promote.yml).

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Add docker promote script and workflow

Taken from @ShubhamTatvamasi's repo (https://github.com/ShubhamTatvamasi/magma-artifactory-docker-promote/blob/master/.github/workflows/docker-promote.yml).

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Small improvements to the docker promote workflow

- Remove checkout action because we are using `wget` to get the `promote.sh` script.
- Pin the docker login action to version 2.0.0.
- Hand variables to `promote.sh` via environment variables, check at the beginning of the script that they exist.
- Add an echo response at the end of the inner loop in `promote.sh` to log out which artifacts have been promoted.
- Disable a shellcheck check.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Improved a logging string

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Added double quotes

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Improved docker-promote workflow

- Added versions as inputs
- Added check if user is a member of "approvers-ci"
- fixed incorrect path to promote script

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Improved docker-promote workflow

- Added version as input
- Added check if user is a member of "approvers-ci"

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Fixed version pinning

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Fixed version pinning

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Simplified use of inputs

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Improved the naming of some variables

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Added license headers

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* feat(ci): Remove default values in promote workflows.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

sebathomas committed:
fix(subscriberdb): Detach inactive subscribers when syncing (#14024)

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

rsarwad committed:
fix(agw): Port enb context from C structure to protobuf structure in s1ap module (#13728)

* fix(agw): Added code to replace std:map wth protobuf map
* fix(agw): Move the c files to cpp and fixed compilation errors
* Changed the extension of header files to hpp in s1ap module
* Modified code to replace hash list with protobuf map for s1ap module
* Moved the map's unit test cases to common library
* Replaced hashlist of UEs within eNB description to map in s1ap test module
* Adding new file mme_init.hpp file
* fix(agw): Modified code to replace hashlist with protobuf map for enb_description_t in s1ap module
* fix(agw): Migrated hash table,mmeid2associd to protobuf map
* Modified code to convert mme_ue_id_imsi_htbl to protobuf map
* fix(agw): Migrate hash list of enb_description_t to protobuf_map
* Modified code to use state_ue_map, declared new parameter in s1ap state manager
* agw(fix): coverted C structure of ue context, ue_description_t to protobuf structure, UeDescription in s1ap module
* Defined S1ap UE state in proto file and removed the enum definition
from s1ap_types.hpp
* Modified code to convert C structure to protobuf structure for enb context

Signed-off-by: Rashmi <rashmi.sarwad@radisys.com>

nstng committed:
chore: cache key of bazel base and image tag are updated (#14041)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

nstng committed:
chore: pip performance is improved by ipv4 over ipv6 precedence (#14057)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Commit:
chore(ci): enable manual execution of the integ test magma_deb in ci (#14076)

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

tmdzk committed:
Upgrading magma version to 1.8.0 (#13869)

Signed-off-by: tmdzk <timodzik@gmail.com>

Signed-off-by: tmdzk <timodzik@gmail.com>
Co-authored-by: tmdzk <timodzik@gmail.com>

mpfirrmann committed:
docs(agw): start-up procedure for containerized AGW is improved (#14077)

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

alexzurbonsen committed:
chore: delete non-existing directories from dockerignore file (#14081)

Signed-off-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>

Signed-off-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>

mpfirrmann committed:
chore: deprecated debian stretch specific files and scripts are removed (#14020)

* chore: deprecated packer templates for libvirt images are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: deprecated packer images for debian machines are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: deprecated aws images using debian are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: deprecated `stretch_snapshot` ansible role is removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: deprecated `provision_stretch.sh` file is removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: several deprecated debian hints are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: packer templates for Ubuntu 16.04 xenial are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: temprary packer template for broken OVS are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: incomplete packer template is removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: packer script for magma-dev vm is renamed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: `setup.sh` since all steps are included in `ubuntu_setup.sh`

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: next batch of debian scripts are removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: hints to debian OS are removed from Ansible roles

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: unused `envoy` Ansible role is removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: deprecated `uselocalpkgrepo` Ansible role is removed

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: readabilty of `magma-dev-virtualbox.json` is improved

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

* chore: hints to old `magma_dev_focal` are adapted

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

Signed-off-by: Marco Pfirrmann <marco.pfirrmann@tngtech.com>

MoritzThomasHuebner committed:
fix(lte): Fixed `detect_init_system` (#14034)

* fix(lte): Fixed `detect_init_system`

- Added `exec_command_capture_output` to have an option to capture stdout and continue on non-zero exit codes.
- Removed `_is_installed` function.
- Simplified docker/systemd_magmad_running logic.
- Renamed `MagmadUtil._data` to `MagmadUtil._credentials` and removed "command" key.
- No longer write "command" to `MagmadUtil._credentials`, as this was not used anyway.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* fix(lte): Added _is_installed check back in

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* fix(lte): Improve changes in previous commits

- replace `&` with `and` in magmad running checks so that second condition is only evaluated if the first suceeds
- Improved docstring of `exec_command_capture_output`
- Added type hints to `exec_command`, `exec_command_output`, and `exec_command_capture_output` to highlight how these methods are different

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* fix(lte): Improve changes in previous commits

- Perform sequential checks for installation and if magma is running
- Remove exec_command_capture_output again as it is now unused

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* Revert "fix(lte): Improve changes in previous commits"

This reverts commit 2bb7908883b1ee396504264835dc60e3e4b335ac.

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

* fix(lte): Properly moved installation checks first

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

Signed-off-by: Moritz Huebner <moritz.huebner@tngtech.com>

sebathomas committed:
fix(agw): Do not check for health of "shared_mconfig" (#14082)

The concept of shared mconfig was added on the top level of the
mconfig a while ago to handle configuration that is used in all
services. However, the health cli tool is using the top-level
entries of mconfig to decide which services to check. the shared
mconfig needs to be excluded here.

Signed-off-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

lucasgonze committed:
chore: Bump node-fetch (#13857)

Signed-off-by: Lucas Gonze <lucas@gonze.com>

paulocalaes committed:
fix(nms): statistics not showing for non super users (#14079)

Signed-off-by: Paulo Henrique Calaes Oliveira <paulo@radtonics.com>
Co-authored-by: Paulo Henrique Calaes Oliveira <paulo@radtonics.com>
Co-authored-by: Sebastian Thomas <sebastian.thomas@tngtech.com>

wolfseb committed:
chore(agw): reduce python container build context (#14085)

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

voisey committed:
docs: Fix double stop of magmad in Quick Start Guide (#14001)

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

wolfseb committed:
docs: cut docs for 1.8.0 release (#14065)

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

voisey committed:
docs: Fix typo in FeG build section (#14088)

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

Signed-off-by: Cameron Voisey <cameron.voisey@tngtech.com>

alexzurbonsen committed:
chore(ci): s1ap integ tests against dockerized agw (#13884)

* chore(ci): add workflow that runs s1ap integ tests against dockerized agw running on magma_dev VM

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>
Signed-off-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>

* chore(ci): correct spelling of set up

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

* chore(ci): shorten and clarify docstring

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

* chore(ci): improve whitespace consistency

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

* chore(ci): remove debug output

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

Signed-off-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>
Signed-off-by: Alexander zur Bonsen <alexander.zur.bonsen@tngtech.com>
Co-authored-by: Fritz Lehnert <13189449+Neudrino@users.noreply.github.com>

Sathyaj27 committed:
fix(amf): Fixing session ambr value when policy attached (#14098)

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

nstng committed:
fix: vagrant box cache location is pointing to correct folder (#14092)

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

wolfseb committed:
fix(agw): fix always true asserts in s1ap integ tests (#14091)

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

pruthvihebbani committed:
Merge remote-tracking branch 'upstream/master' into cpp_migration_nas_ies

pruthvihebbani committed:
Merge branch 'cpp_migration_nas_ies' of https://github.com/pruthvihebbani/magma into cpp_migration_nas_ies

pruthvihebbani committed:
Revert "Update gtpv2c_ie_formatter.c"

This reverts commit 1125e5b62c685bfaf8097a2d2c4eb285222f3b89.

pruthvihebbani committed:
Revert "Update gtpv2c_ie_formatter.c"

This reverts commit 1125e5b62c685bfaf8097a2d2c4eb285222f3b89.

pruthvihebbani committed:
Fixed oai errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Fixed oai errors

Signed-off-by: Pruthvi Hebbani <pruthvi.hebbani@radisys.com>

pruthvihebbani committed:
Merge branch 'cpp_migration_nas_ies' of https://github.com/pruthvihebbani/magma into cpp_migration_nas_ies

pruthvihebbani committed:
Merge remote-tracking branch 'upstream/master' into cpp_migration_nas_ies
